saga/src/main/java/com/iluwatar/saga/choreography/SagaApplication.java

```diff
@@ -0,0 +1,77 @@
+/*
+ * The MIT License
+ * Copyright © 2014-2019 Ilkka Seppälä
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package com.iluwatar.saga.choreography;
+
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * This pattern is used in distributed services to perform a group of operations atomically.
+ * This is an analog of transaction in a database but in terms of microservices architecture this is executed
+ * in a distributed environment
+ * <p>
+ * A saga is a sequence of local transactions in a certain context. If one transaction fails for some reason,
+ * the saga executes compensating transactions(rollbacks) to undo the impact of the preceding transactions.
+ * <p>
+ * In this approach, there are no mediators or orchestrators services.
+ * All chapters are handled and moved by services manually.
+ * <p>
+ * The major difference with choreography saga is an ability to handle crashed services
+ * (otherwise in choreography services very hard to prevent a saga if one of them has been crashed)
+ *
+ * @see com.iluwatar.saga.choreography.Saga
+ * @see Service
+ */
+public class SagaApplication {
+    private static final Logger logger = LoggerFactory.getLogger(SagaApplication.class);
+
+    public static void main(String[] args) {
+        ServiceDiscoveryService sd = serviceDiscovery();
```

Comment:
Note that in Java 11 we can use local variable type inference `var` for most local variables. Please check this throughout the code.

Reply:
I know :). But my laptop is used for work also and the major version is 8 and i have not installed 11 yet. :) If it is critical i will do that  otherwise it would be nice to leave it as it is

Reply:
Ok, understood. We can merge it this way and add it to the list of patterns in #984 and #987